### PR TITLE
feat: Added project create command with `--name`, `--index`, and auto-creation

### PR DESCRIPTION
**Pull request by DraPraks**
Implements #722 

#### Changes made
- `project create` command: New primary command for creating projects with options:
  - `--name` - Specify custom project name
  - `--language` - Explicitly set programming languages (can be used multiple times)
  - `--index` - Index the project immediately after creation
  
- `project index` auto-creation: The `project index` command now automatically creates a project configuration if it doesn't exist (with optional `--name` and `--language` flags for custom setup)

- `project generate-yml` deprecated: Marked for deprecation with user-friendly warning message. Still functional for backward compatibility.

#### Implementation Details
- Extracted shared `_create_project()` helper method to eliminate code duplication between `create` and `index` commands
- All user-facing exceptions converted to `click.ClickException` for error handling
- Race condition handling in auto-creation logic

#### Testing & Quality
- Added test suite (`test_cli_project_commands.py`) with 19 tests covering:
  - Basic project creation with various options
  - Language auto-detection vs. explicit specification
  - `--index` flag functionality
  - Command equivalence verification (`project create --index` ~ `project index`)
  - Backward compatibility of deprecated command
  - Error handling and edge cases
- All tests passing (19/19)
- Code formatted and type-checked using poe

#### Documentation
Updated 3 documentation files with the new command syntax and options:
- 020_running.md
- 040_workflow.md
- 050_configuration.md

Backward Compatibility maintained : `project generate-yml` still works with deprecation notice

#### Files Changed
- cli.py
- test_cli_project_commands.py 
- 020_running.md 
- 040_workflow.md 
- 050_configuration.md 

Happy to make my first contrib to Serena, lmk if any changes are needed.

### CODE REVIEW

File: src/serena/cli.py [hunk 1]
```diff
@@ -431,20 +437,89 @@ def generate_yml(project_path: str, language: tuple[str, ...]) -> None:
...
+        except ValueError as e:
+            raise click.ClickException(str(e))
+
+    @staticmethod
+    @click.command("generate-yml", help="[DEPRECATED] Generate a project.yml file. Use 'project create' instead.")
+    @click.argument("project_path", type=click.Path(exists=True, file_okay=False), default=os.getcwd())
+    @click.option(
+        "--language", type=str, multiple=True, help="Programming language(s); inferred if not specified. Can be passed multiple times."
+    )
+    def generate_yml(project_path: str, language: tuple[str, ...]) -> None:
```
Comment (MischaPanch) [inline src/serena/cli.py]: We can remove this immediately, no need to keep the command. Backwards compatibility was important for the regularly used commands, this is not one of them

File: docs/02-usage/040_workflow.md [hunk 2]
```diff
@@ -22,20 +22,24 @@ You can create a project either
...
    (e.g., `--language python`). 
  * For an existing project, the main programming language will be detected automatically,
    but you can choose to explicitly specify multiple languages by passing the `--language` parameter
    multiple times (e.g. `--language python --language typescript`).
+ * You can optionally specify a custom project name with `--name "My Project"`.
+ * You can immediately index the project after creation with `--index`.
 
 After creation, you can adjust the project settings in the generated `.serena/project.yml` file.
 
+**Note:** The old `project generate-yml` command is deprecated but still functional for backward compatibility.
```
Comment (MischaPanch) [inline docs/02-usage/040_workflow.md]: Also remove pls

File: docs/02-usage/050_configuration.md [hunk 3]
```diff
@@ -22,13 +22,13 @@ Serena is configured in four places:
...
-   generate it explicitly with
+   create it explicitly with
 
    ```shell
-   uvx --from git+https://github.com/oraios/serena serena project generate-yml
+   uvx --from git+https://github.com/oraios/serena serena project create
    ```
 
-   (or use the `--directory` command version).
+   You can optionally specify the project name (`--name`), languages (`--language`), and immediately index (`--index`).
```
Comment (MischaPanch) [inline docs/02-usage/050_configuration.md]: pls just write `... create [options]`, repeated documentation is difficult to maintain. Already here it's diverging from the docs in workflow.md, where we expain how to pass multiple languages

Comment (opcode81) [reply docs/02-usage/050_configuration.md]: I think this is fine here. It's an important command and unlikely to change much in the future.
Which divergence are you referring to?
(We do need to name important things in the docs, even if it means a bit more work to maintain it, and we should probably even extend the document on configuration to mention more things explicitly - but not in this PR.)

Comment (MischaPanch) [reply docs/02-usage/050_configuration.md]: In the other place we tell the user how to pass multiple languages (using `--language l1 --language l2`), it's not super intuitive

Comment (opcode81) [reply docs/02-usage/050_configuration.md]: But there is no divergence. Here, too, it says "languages" (plural); how exactly to pass multiple is not said. (But it probably should be, which is why I think we should say more about usage, not less, in the docs.)

Comment (opcode81) [reply docs/02-usage/050_configuration.md]: That's why, when we talked about the docs PR, I specifically said that the "configuration" and "workflow" pages still need work. We need to explain the options better and make everything clear.

Comment (DraPraks) [reply docs/02-usage/050_configuration.md]: Should I just put something like this in 050_configuration.md then? :

`serena project create [options]` (see the [Project Workflow](040_workflow.md#project-creation--indexing) 
   for details on available options).
   
 So it just links straight to the workflow docs

Comment (MischaPanch) [reply docs/02-usage/050_configuration.md]: Sounds good to me, wdyt @opcode81 ?

Comment (opcode81) [reply docs/02-usage/050_configuration.md]: Yes, let's go with that for now.

**Comment by MischaPanch** [comment]
Thanks, merging! We might change some things around the tests or documentation, but that's easiest done post merge

**Comment by DraPraks** [comment]
Sounds good, thanks 😄
